File: hooks/search-word/csv/src/parser/field.rs
use crate::csv_value::CSVValue;

use super::{escaped::escaped, non_escaped::non_escaped, ParseResult};

pub fn field<'a>(s: &'a str) -> ParseResult<'a, CSVValue> {
  let first = s.chars().next().unwrap();

  if first == '"' {
    escaped(&s[1..])
  } else {
    non_escaped(s)
  }
}

#[cfg(test)]
mod test_field {
    use crate::{csv_value::CSVValue, parser::field::field};

  #[test]
  fn test_field(){
    let str0 = "3.14,sample1";
    let str1 = "\"sample0\",sample1";

    assert_eq!(field(str0), (CSVValue::Float(3.14), ",sample1"));
    assert_eq!(field(str1), (CSVValue::String("sample0".to_string()), ",sample1"));
  }
}

File: hooks/csv/field.ts
import escaped from './escaped';
import nonEscaped from './nonEscaped';
import { ParseResult } from './parser';

const field = (s: string): ParseResult<string> => {
  if (s[0] == '"') {
    return escaped(s.slice(1));
  } else {
    return nonEscaped(s.slice(1));
  }
};

export default field;
